src/message.c

```diff
@@ -2625,59 +2625,53 @@ void qd_message_stream_data_release(qd_message_stream_data_t *stream_data)
 
     qd_message_pvt_t     *pvt       = stream_data->owning_message;
     qd_message_content_t *content   = pvt->content;
-    qd_buffer_t          *buf;
 
     //
     // find the range of buffers that do not overlap other stream_data
     // or msg->body_buffer
     //
 
-    qd_buffer_t *start_buf = stream_data->free_prev ? DEQ_PREV(stream_data->section.buffer) : stream_data->section.buffer;
-    if (DEQ_PREV(stream_data) && DEQ_PREV(stream_data)->last_buffer == start_buf) {
-        // overlap previous stream_data
-        if (start_buf == stream_data->last_buffer) {
-            // no buffers to free
-            DEQ_REMOVE(pvt->stream_data_list, stream_data);
-            free_qd_message_stream_data_t(stream_data);
-            return;
-        }
+    qd_buffer_t *stop_buf = stream_data->last_buffer;
+    qd_buffer_t *end_boundary = DEQ_NEXT(stream_data) ? DEQ_NEXT(stream_data)->first_buffer : pvt->body_buffer;
+    if (stop_buf != end_boundary)
+        // safe to free last_buffer since this message no longer references it
+        stop_buf = DEQ_NEXT(stop_buf);
+    assert(stop_buf);
```

Comment:
Should this assert check for stop_buf == end_boundary ?

Reply:
No, stop_buf never gets free (it's where we "stop" freeing buffers) so stop_buf == end_boundary is legit and will happen often.